### DISCUSSION TITLE
Whitelist for bypassing number of clients

### DISCUSSION THREAD

**Discussion by rdica** [jamulussoftware/jamulus#2889] [Ideas]
Request for a username or IP based whitelist that allows connections regardless if max number of clients is reached.

I came across a use case where one of the regular players would get disconnected a few times during a session, but couldn't rejoin due to max clients connected, unless I kicked everyone off by restarting the server, or dropping non-participants via manual firewall changes.
Increasing the number of clients isn't an option.

**Comment by pljones** [comment]
I think this wouldn't be a simple change, if I'm remembering things rightly -- the server would need to be restarted to change the number of available channels.

**Comment by dcorson-ticino-com** [comment]
Why is increasing the number of clients not an option when that is exactly what you are asking for?

**Reply by rdica** [reply]
Except its not exactly what I'm asking for. I'm asking for a whitelist thats exempted from the limit so a core group of players wouldn't be denied connection when the max is reached. Simply increasing the number of clients does not guarantee whitelisted players would still be able to join, in fact my experience with increasing the number of clients brought more problems than it resolved, and still resulted in desired people being unable to join. There's no desire to go private as we still want other people besides our core to join us.

**Reply by dcorson-ticino-com** [reply]
How would you then decide which players get thrown out of the session if you don't increase the number of clients ?
(I understand and accept your problem, I just don't see a good way to solve it)

**Reply by rdica** [reply]
Currently one would either have to restart the server to drop everyone, and hope the core joins before the limit's hit, or manually firewall off IP addresses of non-participants in the hopes that desired participants keep trying to reconnect until an open slot becomes available.

I can certainly understand if there isn't a good way to implement such a request. I had to ask :)

I know there's been past discussions regarding server management, access control, etc. The introduction of the new JSON-RPC API could certainly provide management capabilities for these new feature requests.

**Reply by ghost** [reply]
The logic to this is simple. All you need extra; is a whitelist that reserves seats from the maximum number of clients. These settings would, at the very least,  be applied before the server is running. I think adding a whitelist is a good idea, given that lesser features are already implemented.
